selendroid-server/src/main/java/io/selendroid/server/model/DefaultSelendroidDriverFactory.java

```diff
@@ -4,9 +4,22 @@
 
 public class DefaultSelendroidDriverFactory implements SelendroidDriverFactory {
 
+    private static SelendroidDriver selendroidDriver = null;
+    private static final String SELENDROID = "selendroid";
+    private static final String UIAUTOMATOR = "uiautomator";
+
     @Override
-    public SelendroidDriver createSelendroidDriver( ServerInstrumentation androidInstrumentation ) {
-        return new DefaultSelendroidDriver(androidInstrumentation);
+    public SelendroidDriver createSelendroidDriver(ServerInstrumentation androidInstrumentation, String automationName) {
+        if (selendroidDriver == null) {
```

Comment:
Why isn't this code just:

```java
if (automationName.equals(UIAUTOMATOR)) {
  return new DefaultAppiumUiAutomatorDriver(androidInstrumentation);
} else {
  return new DefaultSelendroidDriver(androidInstrumentation);
}
```

The singleton logic around selendroidDriver seems wrong since it's never assigned a value.